sr_wear_logger/test/test_sr_wear_logger.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python
+
+# Copyright 2021 Shadow Robot Company Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the Free
+# Software Foundation version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import rospy
+import rospkg
+import rostest
+from unittest import TestCase
+from std_msgs.msg import Bool
+from sr_utilities_common.manual_test_suite import ManualTestSuite
+from sr_utilities_common.aws_manager import AWS_Manager
+from sr_wear_logger.sr_wear_logger import SrWearLogger
+from sr_wear_logger.dummy_publisher import DummyPublisher
+import os
+import re
+import yaml
+
+
+class TestSrWearLogger(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.path_to_test_folder = rospkg.RosPack().get_path('sr_wear_logger') + "/test"
+        cls.path_to_test_file = cls.path_to_test_folder + "/wear_data.yaml"
+        cls.log_file_key_list = ["total_time_[s]", "total_angles_[rad]"]
+
+    @classmethod
+    def tearDownClass(cls):
+        for file in os.listdir(cls.path_to_test_folder):
+            pattern = "\d\d\d\d-\d\d-\d\d-\d\d-\d\d-\d\d.yaml"
+            if bool(re.match(pattern, file)):
+                os.remove(cls.path_to_test_folder + "/" + file)
+
+    @classmethod
+    def check_are_values_numeric(self, dictionary):
+        for k, v in dictionary.items():
+            if isinstance(v, dict):
+                dictionary[k] = self.check_are_values_numeric(v)
+            else:
+                if not(type(dictionary[k]) is float or type(dictionary[k]) is int):
+                    return False
+        return True
+
+    def test_10_check_param_case_wrong_hand_serial(self):
+        test_wear_logger = SrWearLogger("", 1, 1)
+        self.assertFalse(test_wear_logger.check_parameters())
+
+    def test_11_check_param_case_wrong_time(self):
+        test_wear_logger = SrWearLogger("test", 1.2, 1)
+        self.assertFalse(test_wear_logger.check_parameters())
+
+    def test_12_check_param_case_all_correct(self):
+        test_wear_logger = SrWearLogger("test", 10, 2)
+        self.assertTrue(test_wear_logger.check_parameters())
+
+    def test_13_check_log_file_existence(self):
+        test_wear_logger = SrWearLogger("test", 10, 2)
```

Comment:
same as above, if we can wait a bit less (e.g. 5 seconds it might be good)

Reply:
Sure